### ISSUE TITLE
Fix package/resource location in example tests

### ISSUE THREAD

**Issue by nrmancuso** [checkstyle/checkstyle#13484]
From https://github.com/checkstyle/checkstyle/pull/13482#discussion_r1278313631:
>We have a bunch of other example tests that we merged that need to be fixed:
```
➜  checkstyle git:(master) ag -Q 'return "xdocs-examples";' -B 1 | grep 'getPackageLocation' | wc -l
14


```
```
➜  checkstyle git:(master) ✗  ag -Q 'return "xdocs-examples";' -B 1 | grep 'getPackageLocation' | cut -d'-' -f 2
examples/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationOnSameLineCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationUseStyleCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/annotation/MissingDeprecatedCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/annotation/MissingOverrideCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/annotation/SuppressWarningsCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyCatchBlockCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/blocks/AvoidNestedBlocksCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/blocks/NeedBracesCheckExamplesTest.java:30
examples/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheckExamplesTest.java:30

```

**Comment by rnveach** [comment]
Context:
> Looks like we have getPackageLocation and getResourceLocation confused

**Comment by rnveach** [comment]
> is there an easy to validate this?
> Can't think of anything else besides running the tests/adding some xpath check to ensure that getResourceLocation is xdocs-examples.

We should make `AbstractExamplesModuleTestSupport` similar to `AbstractCheckstyleModuleTestSupport` and `AbstractItModuleTestSupport` and others, and have all examples extend this.

**Comment by nrmancuso** [comment]
Closed via #13489
